crates/polars-core/src/frame/group_by/aggregations/dispatch.rs

```diff
@@ -164,15 +174,26 @@ impl Series {
             Boolean => self.cast(&Float64).unwrap().agg_mean(groups),
             Float32 => SeriesWrap(self.f32().unwrap().clone()).agg_mean(groups),
             Float64 => SeriesWrap(self.f64().unwrap().clone()).agg_mean(groups),
-            dt if dt.is_numeric() => {
-                apply_method_physical_integer!(self, agg_mean, groups)
-            },
-            dt @ Duration(_) => {
-                let s = self.to_physical_repr();
-                // agg_mean returns Float64
-                let out = s.agg_mean(groups);
-                // cast back to Int64 and then to logical duration type
-                out.cast(&Int64).unwrap().cast(dt).unwrap()
+            dt if dt.is_numeric() => apply_method_physical_integer!(self, agg_mean, groups),
+            #[cfg(feature = "dtype-datetime")]
+            dt @ Datetime(_, _) => self
+                .to_physical_repr()
+                .agg_mean(groups)
+                .cast(&Int64)
+                .unwrap()
+                .cast(dt)
+                .unwrap(),
+            dt @ (Date | Duration(_) | Time) => {
+                let ca = self.to_physical_repr();
+                let physical_type = ca.dtype();
+                let s = apply_method_physical_integer!(ca, agg_mean, groups);
+                if dt.is_logical() {
```

Comment:
this branch is always true. So the `if` can be elided.

The same is true for L128

Reply:
Thanks @ritchie46, fixing now. Out of curiosity, do I have to split these into separate branches if I want to properly feature-gate each of them?

Reply:
I am not entirely sure what you mean. :sweat_smile: What do you want to feature gate?

Reply:
Ahh--I meant match on `Date | Duration(_) | Time`--should these each be behind the `dtype-date`, `dtype-duration`, and `dtype-time` feature gates?

Reply:
Ah.. Yes the  you split them out and put the branch logic in a function.

Reply:
Ok thanks. I'll include that in my next PR (which will touch this portion anyway).